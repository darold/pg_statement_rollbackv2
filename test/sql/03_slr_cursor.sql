\set VERBOSITY default
\set ECHO all
LOAD 'pg_statement_rollback.so';
SET pg_statement_rollback.enabled = 0;

DROP SCHEMA IF EXISTS testrsl CASCADE;
CREATE SCHEMA testrsl;

SET search_path TO testrsl,public;

\echo Setup table
BEGIN;
    DROP TABLE IF EXISTS tbl_rsl;
    CREATE TABLE tbl_rsl(id integer, val varchar(256));
    INSERT INTO tbl_rsl VALUES (1, 'one');
COMMIT;

\echo Check behavior when pg_statement_rollback.enabled is disabled
BEGIN;
DECLARE c CURSOR FOR SELECT id FROM tbl_rsl ORDER BY id;
SAVEPOINT one;
FETCH 1 FROM c;
ROLLBACK TO SAVEPOINT one;
FETCH 1 FROM c;
RELEASE SAVEPOINT one;
FETCH 1 FROM c;
CLOSE c;
DECLARE c CURSOR FOR SELECT id/0 FROM tbl_rsl ORDER BY id;
SAVEPOINT two;
FETCH 1 FROM c;
ROLLBACK TO SAVEPOINT two;
FETCH 1 FROM c;
ROLLBACK TO SAVEPOINT two;
RELEASE SAVEPOINT two;
FETCH 1 FROM c;
ROLLBACK;

DROP SCHEMA testrsl CASCADE;
