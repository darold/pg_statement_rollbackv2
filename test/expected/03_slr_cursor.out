\set VERBOSITY default
\set ECHO all
LOAD 'pg_statement_rollback.so';
SET pg_statement_rollback.enabled = 0;
DROP SCHEMA IF EXISTS testrsl CASCADE;
NOTICE:  schema "testrsl" does not exist, skipping
CREATE SCHEMA testrsl;
SET search_path TO testrsl,public;
\echo Setup table
Setup table
BEGIN;
    DROP TABLE IF EXISTS tbl_rsl;
NOTICE:  table "tbl_rsl" does not exist, skipping
    CREATE TABLE tbl_rsl(id integer, val varchar(256));
    INSERT INTO tbl_rsl VALUES (1, 'one');
COMMIT;
\echo Check behavior when pg_statement_rollback.enabled is disabled
Check behavior when pg_statement_rollback.enabled is disabled
BEGIN;
DECLARE c CURSOR FOR SELECT id FROM tbl_rsl ORDER BY id;
SAVEPOINT one;
FETCH 1 FROM c;
 id 
----
  1
(1 row)

ROLLBACK TO SAVEPOINT one;
FETCH 1 FROM c;
 id 
----
(0 rows)

RELEASE SAVEPOINT one;
FETCH 1 FROM c;
 id 
----
(0 rows)

CLOSE c;
DECLARE c CURSOR FOR SELECT id/0 FROM tbl_rsl ORDER BY id;
SAVEPOINT two;
FETCH 1 FROM c;
ERROR:  division by zero
ROLLBACK TO SAVEPOINT two;
FETCH 1 FROM c;
ERROR:  portal "c" cannot be run
ROLLBACK TO SAVEPOINT two;
RELEASE SAVEPOINT two;
FETCH 1 FROM c;
ERROR:  portal "c" cannot be run
ROLLBACK;
DROP SCHEMA testrsl CASCADE;
NOTICE:  drop cascades to table tbl_rsl
